gulpfile.js

```diff
@@ -0,0 +1,96 @@
+const autoprefixer = require('gulp-autoprefixer');
+const bs = require('browser-sync').create('amagaki-browsersync');
+const esbuild = require('esbuild');
+const gulp = require('gulp');
+const sass = require('gulp-sass');
+const {exec} = require('child_process');
+sass.compiler = require('sass');
+
+const ENTRIES = {
+  js: {
+    // File location for tsc output.  Based on tsconfig output settings.
+    tsc_out: ['./dist/ts/main.js'],
+    out: './dist/js/main.min.js',
+    watch: ['./src/ts/**/*.ts'],
+  },
+  sass: {
+    src: ['./src/sass/**/*.sass'],
+    out: './dist/css/',
+    watch: ['./src/sass/**/*.sass'],
+  },
+};
+
+/**
+ * esBuild does not do type checks and can build with type errors so we first run
+ * `tsc` and generate a JS file.  esBuild is then run on the outputted JS file.
+ *
+ * The entry point of tsc compilation is configured in tsconfig `include`.
+ */
+const runEsBuild = async prod => {
+  return new Promise(resolve => {
+    exec('tsc', async (error, stderr) => {
+      if (stderr) {
+        console.error('Typescript errors');
+        console.error(stderr);
+        resolve();
```

Comment:
i think you should use reject() here instead so that the gulp task is marked as failed.